src/server/action_process_queue.ts

```diff
@@ -0,0 +1,52 @@
+import * as spawn from "child_process"
+import * as express from "express"
+import * as Pq from "p-queue"
+import * as winston from "winston"
+
+export class AsyncProcessJob {
+    queue: Pq
+
+    constructor() {
+        this.queue = new Pq({concurrency: 1})
+    }
+
+    async asyncProcess(req: express.Request, res: express.Response) {
+        return this.queue.add(async () => {
+            return new Promise<void>((resolve, reject) => {
+                const child = spawn.fork(`./src/actions/actions_process.ts`)
+                child.on("message", (actionResponse) => {
+                    // Some versions of Looker do not look at the "success" value in the response
+                    // if the action returns a 200 status code, even though the Action API specs otherwise.
+                    // So we force a non-200 status code as a workaround.
+                    winston.info("Success? " + actionResponse.success
+                        + " || and what: " + JSON.stringify(actionResponse))
+                    if (!actionResponse.success) {
```

Comment:
that this code is duplicated is a good indication it's not factored as well as it could be. https://github.com/looker/actions/pull/192/files#r211036573 should let you clean that up